packages/key-did-resolver/src/secp256k1/secp256k1-key-pair.ts

```diff
@@ -0,0 +1,146 @@
+import crypto from 'crypto'
+
+import bs58 from 'bs58'
+import secp256k1 from 'secp256k1'
+import * as utils from './secp256k1-utils'
+
+/**
+ * Encapsulates Secp256k1 key pair
+ */
+export class Secp256k1KeyPair {
+    public id: string
+    public type: string
+    public controller: string
+
+    public publicKeyBase58: string
+    public privateKeyBase58: string
+
+    constructor(options: any = {}) {
+        this.type = 'EcdsaSecp256k1VerificationKey2019'
+
+        this.id = options.id
+        this.controller = options.controller
+        this.privateKeyBase58 = options.privateKeyBase58
+        this.publicKeyBase58 = options.publicKeyBase58
+    }
+
+    /**
+     * Constructs fingerprint from public key
+     */
+    static fingerprintFromPublicKey({ publicKeyBase58 }: any): string {
+        const pubkeyBytes = bs58.decode(publicKeyBase58)
+        const buffer = new Uint8Array(2 + pubkeyBytes.length)
+        // See https://github.com/multiformats/multicodec/blob/master/table.csv
+        // 0xe7 is Secp256k1 public key
+        buffer[0] = 0xe7 //
+        buffer[1] = 0x01
+        buffer.set(pubkeyBytes, 2)
+        // prefix with `z` to indicate multi-base base58btc encoding
+        return `z${bs58.encode(buffer)}`
+    }
+
+    /**
+     * Derive instance from options
+     */
+    static async from(options: any): Promise<Secp256k1KeyPair> {
+        let privateKeyBase58 = options.privateKeyBase58
+        let publicKeyBase58 = options.publicKeyBase58
+
+        if (options.privateKeyHex) {
```

Comment:
Might be worth adding a check if both `privateKeyHex` and `privateKeyJwk` or `publicKeyHex` and `publicKeyJwk` are provided to throw an error so that only one variant should be provided.

I think it should also check if either private key variant is provided and throw an error if not rather than create an invalid instance, and if the private key is provided but not the public key, shouldn't this function derive it?

Reply:
tnx! removed this piece of code 😊